yb-voyager/cmd/exportData.go

```diff
@@ -788,14 +844,238 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+	var partitionsToRootTableMap map[string]string
+	var err error
+	storedTableList := make([]string, 0)
+	fetchNameTupleFromNameRegDirectly := true
+	switch source.DBType {
+	case ORACLE, MYSQL:
+		storedTableList = msr.TableListExportedFromSource
+	case POSTGRESQL:
+		storedTableList = msr.SourceExportedTableListWithLeafPartitions
+		partitionsToRootTableMap = msr.SourceRenameTablesMap
+		//In case of PG we need to check the leaf condititon for the tables in table list first and then return the nametuple
+		//via namereg or handcrafted one
+		fetchNameTupleFromNameRegDirectly = false
+	case YUGABYTEDB:
+
+		//In case of YB we need to check the leaf condititon for the tables in table list first and then return the nametuple
+		//via namereg or handcrafted one
+		fetchNameTupleFromNameRegDirectly = false
+		// On subsequent run after the first we will use the stored table with leaf partitions
+		storedTableList = msr.TargetExportedTableListWithLeafPartitions
+		partitionsToRootTableMap = msr.TargetRenameTablesMap
+
+		//For the first run of export data from target we use the TableListExportedFromSource (which has only root tables)
+		if len(msr.TargetExportedTableListWithLeafPartitions) == 0 || msr.TargetRenameTablesMap == nil {
+			storedTableList = msr.TableListExportedFromSource
+			if msr.SourceDBConf.DBType != POSTGRESQL {
+				//but in case we are using this source list and its source is not PG then we can directly use the namereg
+				fetchNameTupleFromNameRegDirectly = true
+			}
+
+		}
+
+	}
+
+	for _, table := range storedTableList {
+		tuple, err := getNameTupleFromQualifiedObject(table, nil, fetchNameTupleFromNameRegDirectly)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error getting name  tuple for the string oject name: %v", err)
+		}
+		firstRunTableWithLeafsAndRoots = append(firstRunTableWithLeafsAndRoots, tuple)
+	}
+
+	if source.DBType == YUGABYTEDB {
+		//For the first run of export data from target we will fetch the leaf partitions from target and store them
+		if len(msr.TargetExportedTableListWithLeafPartitions) == 0 || msr.TargetRenameTablesMap == nil {
+			//Now add the leaf partitions to the stored table-list for the first run of export data from target
+			// as it will only have root table names and get the partitionsToRootTableMap
+			partitionsToRootTableMap, firstRunTableWithLeafsAndRoots, err = addLeafPartitionsInTableList(firstRunTableWithLeafsAndRoots, true)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to add the leaf partitions in table list: %w", err)
+			}
+		}
+	}
+	return firstRunTableWithLeafsAndRoots, partitionsToRootTableMap, nil
+}
+
+func getInitialTableList() (map[string]string, []sqlname.NameTuple, error) {
+	// store table list after filtering unsupported or unnecessary tables
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("error fetching migration status record: %v", err)
+	}
+	storedTableListNotAvailable := func() bool {
```

Comment:
shouldn't we handle this for export-data-from-target as well?

Reply:
we can not go to DB for the export-data-from target as such, because we need to use the table list that is used from export (source one), which is available in msr. So, handling export data from target even first run in the below func`retrieveFirstRunListAndPartitionsRoot` which handles target case as well